core/encoding.go

```diff
@@ -119,6 +120,11 @@ func DecodeFieldValue(fieldDesc client.FieldDescription, val any) (any, error) {
 			case uint:
 				return int64(v), nil
 			}
+		case client.FieldKind_DATETIME:
```

Comment:
question: Why has this been added?

Reply:
Because time.Time is saved to the datastore as a string. It needs to be converted back to time.Time.

Reply:
Yes, but it has always been saved as a string, did you find a bug?  If so is there a test(s) that covers the bug?

Reply:
Since we enforce strong typing, we simply convert it back to time.Time once we fetch the field. Otherwise a string is returned to the user instead of the expected time.Time.

Reply:
Ah okay nice - thanks Fred :)